### PR TITLE
React redux components

### PR DESCRIPTION
**Pull request by mullenlucas**
## Implemented features:

- Added the `react-redux` and `uuid` packages
- Modified the initial state in books.js to a sample array of books. Instanciated useSelector() to consume this initial state array to display the books
- Used redux Provider to update information on the App.js components
- Generated unique IDs for each book via the uuid package
- Implemented functionality to add new books by dispatching the action of addBook()
- Implemented functionality to remove books by dispatching the action of removeBook()
- Implemented functionality to check book status by dispatching the action of statusChecker()

### CODE REVIEW

File: src/components/BooksGen.js [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
           <button type="button" className="axn-btn" id="comment">Comments</button>
-          <button type="button" className="axn-btn" id="remo">Remove</button>
+          <button
+            type="button"
+            className="btn"
+            id="comment"
+            onClick={() => disp(removeBook(book))}
+          >
+            Remove
+          </button>
```
Comment (omar-muhamad) [inline src/components/BooksGen.js]: - [x] Clicking on the remove button suppose to dispatch the remove action and remove the book from the page and this is not happining. So please fix that.

chrome-capture-2022-8-28

File: src/display/Categories.js [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
   return (
     <div className="categ-sec">
-      <button type="button" id="check-status">Check status</button>
+      <button
+        type="button"
+        id="check-status"
+        onClick={() => dispatch(statusChecker())}
+      >
+        Check status
+      </button>
```
Comment (omar-muhamad) [inline src/display/Categories.js:16]: - [x] Clicking on the `check status` button suppose to show `under construction` the text data you get from dispatching `statusChecker` action. So please fix that.

chrome-capture-2022-8-28 (1)

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @mullenlucas,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShahierNashaat — APPROVED** [review]
Hi @mullenlucas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**